packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -35,19 +36,24 @@ export class TerminalFrontendContribution implements CommandContribution, MenuCo
 
     constructor(
         @inject(FrontendApplication) protected readonly app: FrontendApplication,
-        @inject(WidgetManager) protected readonly widgetManager: WidgetManager
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService
     ) { }
 
     registerCommands(commands: CommandRegistry): void {
-        commands.registerCommand(TerminalCommands.NEW, {
-            isEnabled: () => true,
-            execute: () => this.newTerminal()
+        this.workspaceService.root.then(() => {
+            commands.registerCommand(TerminalCommands.NEW, {
+                isEnabled: () => true,
```

Comment:
Should we have a single provider of default workspace root for all clients? Instead of each client fallbacks to its default? e.g. if we figure out that for electron `electron.process.cwd` should be used as a default, only one place should be changed.

Reply:
I'm not sure, I actually thought the idea of the PR was to have undefined as default pretty much.
I don't think it makes sense to open cwd as a default workspace, especially given that we inotify this root dir... if the user opens it from $HOME or /usr/bin... this will be bad.

It does make sense for the terminal however. 

In general I don't think the terminal should be linked to the workspace except to provide the convenience of starting in the root dir if there is one.

Reply:
> It does make sense for the terminal to use cwd however.

It won't work nicely for the packaged electron. VS code opens the terminal in the user home dir if a folder is not opened and otherwise in the workspace root.

> In general I don't think the terminal should be linked to the workspace except to provide the convenience of starting in the root dir if there is one.

I thought an idea was that:
- components which can work without an opened project should use `this.tryRoot` and fall back to the common non-project root;
- components, which cannot, should use `this.root` and wait when a project is opened

@kittaakos correct me if I am wrong

Reply:
@akosyakov  OK this makes more sense to me, thx!

Reply:
> It does make sense for the terminal to use cwd however.

Why? You get a terminal to the server's cwd. I don't see why that is better than a workspace root there?

Reply:
I mean that it's possible to use cwd if you don't have a workspace root. Of course it's better to use the workspace root.

Note that I linke @akosyakov idea.. with the tryRoot I think that's nice.